Webapp/sources/blueprints/reaction_list/routes.py

```diff
@@ -98,3 +99,18 @@ def get_schemata() -> Response:
     )
     schemes = services.reaction.make_scheme_list(reaction_list, size)
     return {"schemes": schemes, "sort_crit": escape(sort_crit)}
+
+
+@reaction_list_bp.route("/get_new_reaction_id", methods=["GET", "POST"])
+@login_required
+def get_new_reaction_id() -> Response:
+    workbook_name = request.json['workbook']
+    workgroup_name = request.json['workgroup']
+
```

Comment:
Instead of just accessing the properties like this, I would use `request.json.get("....")`. Then test if they are `None`. If they are `None`, return a `Response` object with the status code of 400 for Bad Request. As it is now, you'd get a `KeyError` which I imagine would show up as a server error (5XX).